libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -34,8 +34,10 @@ namespace solidity
 
 NameAndTypeResolver::NameAndTypeResolver(
 	vector<Declaration const*> const& _globals,
+	map<ASTNode const*, shared_ptr<DeclarationContainer>>& _scopes,
```

Comment:
Not at all, this is C++... If you pass in references, you have to check the documentation to see who will own it.